### PR TITLE
Android OpenCL tutorial

### PR DESCRIPTION
**Pull request by apavlenko**
description for android tutorial4

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
cool! :+1:

File: doc/tutorials/introduction/android_binary_package/android_ocl_intro.markdown [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+    theQueue.enqueueNDRangeKernel(Laplacian, cl::NullRange, cl::NDRange(w, h), cl::NullRange);
+    theQueue.finish();
+
+    theQueue.enqueueReleaseGLObjects(&images);
+    theQueue.finish();
+@endcode
+
+### OpenCV T-API
+
+Bug instead of writing OpenCL code by yourselves you may want to use __OpenCV T-API__ that calls OpenCL implicitly.
```
Comment (PhilLab) [inline doc/tutorials/introduction/android_binary_package/android_ocl_intro.markdown]: Thanks for the tutorial :+1: . Here is a small typo: Bug -> But

File: doc/tutorials/introduction/android_binary_package/android_ocl_intro.markdown [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+public abstract class MyGLRendererBase implements GLSurfaceView.Renderer, SurfaceTexture.OnFrameAvailableListener {
+    protected final String LOGTAG = "MyGLRendererBase";
+
+    protected SurfaceTexture mSTex;
+    protected MyGLSurfaceView mView;
+
+    protected boolean mGLInit = false;
+    protected boolean mTexUpdate = false;
+
+   MyGLRendererBase(MyGLSurfaceView view) {
```
Comment (alalek) [inline doc/tutorials/introduction/android_binary_package/android_ocl_intro.markdown]: indentation issue

**Comment by alalek** [comment]
@apavlenko Please squash commits into one (to eliminate unnecessary intermediate changes)

**Comment by apavlenko** [comment]
Makes sense. Done.
(just to mention: this PR has relation to #4239)

**Comment by alalek** [comment]
Thanks!
:+1:

**Pull request merged**